beta/src/content/docs/en/guides/server-side-rendering.md

```diff
@@ -0,0 +1,289 @@
+
+const notifyUserFx;
+
+const $counter = createStore<number | null>(null);
+
+const fetchUserCounterFx = createEffect(async () => {
+  await sleep(100); // in real life it would be some api request
+
+  return Math.floor(Math.random() * 100);
+});
+
+const buttonClicked = createEvent();
+const saveUserCounterFx = createEffect(async (count: number) => {
+  await sleep(100); // in real life it would be some api request
+});
+
+sample({
+  clock: appStarted,
+  source: $counter,
+  filter: (count) => count === null, // if count is already fetched - do not fetch it again
+  target: fetchUserCounterFx,
+});
+
+sample({
+  clock: fetchUserCounterFx.doneData,
+  target: $counter,
+});
+
+sample({
+  clock: buttonClicked,
+  source: $counter,
+  fn: (count) => count + 1,
+  target: [$counter, saveUserCounterFx],
+});
+
+const $countUpdatePending = combine(
+  [fetchUserCounterFx.pending, saveUserCounterFx.pending],
+  (updates) => updates.some((upd) => upd === true),
+);
+
+const $isClient = createStore(typeof document !== "undefined", { serialize: "ignore" });
+
+const notifyFx = createEffect((message: string) => {
+  alert(message);
+});
+
+sample({
+  clock: [
+    saveUserCounterFx.done.map(() => "Counter update is saved successfuly"),
+    saveUserCounterFx.fail.map(() => "Could not save the counter update :("),
+  ],
+  // It is totally ok to have some splits in the app's logic based on current environment
+  //
+  // Here we want to trigger notification alert only at the client
+  filter: $isClient,
+  target: notifyFx,
+});
+
+// ui
+export function App() {
+  const clickButton = useUnit(buttonClicked);
+  const { count, updatePending } = useUnit({
+    count: $counter,
+    updatePending: $countUpdatePending,
+  });
+
+  return (
+    <div>
+      <h1>Counter App</h1>
+      <h2>{updatePending ? "Counter is updating" : `Current count is ${count ?? "unknown"}`}</h2>
+      <button onClick={() => clickButton()}>Update counter</button>
+    </div>
+  );
+}
+```
+
+This is our app's code which will be used to both server-side render and to handle client's needs.
+
+## Server entrypoint
+
+The way of the `<App />` to the client browsers starts at the server. For this we need to create **separate entrypoint** for the specific server-related code, which will also handle the server-side render part.
+
+In this example we're not going to dive deep into various possible server impelementations - we will focus on the request handler itself instead.
+
+:::info
+Alongside with basic SSR needs, like calculating the final state of the app and serializing it, `effector` also handles **the isolation of user's data between requests**
+
+It is very important feature, as `Node.js` servers usually handle more than one user request at the same moment of time.
+
+Since JS-based platforms , including `Node.js`, usually have single "main" thread - all logical computaions are happening in the same context, with the same memory available.
```

Comment:
```suggestion
Since JS-based platforms, including `Node.js`, usually have single "main" thread - all logical computaions are happening in the same context, with the same memory available.
```